cmake/vif.cmake

```diff
@@ -4,14 +4,33 @@
 # Generates USB-C VIF policies in XML format from device tree.
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
-set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+set(vif_xml vif.xml)
+set(vif_out ${PROJECT_BINARY_DIR}/${vif_xml})
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
-        --vif-out ${vif_xml}
-        --board ${BOARD}
+        --vif-out ${vif_out}
         )
 
+if(CONFIG_GENVIF_INPUT_VIF_XML_PATH)
+    if (EXISTS ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+    elseif (EXISTS ${BOARD_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        set(vif_source_xml ${BOARD_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+    endif()
```

Comment:
Is the priority inversion compared to the other case intentional?

Reply:
Cool, but why?

Reply:
I removed board from config option lookup, as it seems be to not required because the board code is generic for all apps. At present if kconfig option is configured then it will check for that path in application dir. If option is not enabled it will check for vif.xml in board or app directories. The order will enable if vif.xml is defined at both app and board level. The app will have the choice to choose.